lib/AdapterLdap.php

```diff
@@ -340,4 +340,43 @@ public function getMemberStatusByUserAndVo($user, $vo)
         }
         return Member::VALID;
     }
+
+    public function getResourceCapabilities($entityId, $userGroups)
+    {
+        $facility = $this->connector->searchForEntity(
+            $this->ldapBase,
+            '(&(objectClass=perunFacility)(entityID=' . $entityId . '))',
+            ['perunFacilityId']
+        );
+        $facilityId = $facility['perunFacilityId'][0];
+
+        $resources = $this->connector->searchForEntities(
+            $this->ldapBase,
+            '(&(objectClass=perunResource)(perunFacilityDn=perunFacilityId=' . $facilityId . ','
+            . $this->ldapBase . '))',
+            ['capabilities', 'assignedGroupId']
+        );
+
+        $userGroupsIds = [];
+        foreach ($userGroups as $userGroup) {
+            array_push($userGroupsIds, $userGroup->getId());
+        }
+
+        $resourceCapabilities = [];
+        foreach ($resources as $resource) {
+            if (
+                !array_key_exists('assignedGroupId', $resource) ||
+                !array_key_exists('capabilities', $resource)
+            ) {
+                continue;
+            }
+            foreach ($resource['assignedGroupId'] as $groupId) {
+                if (in_array($groupId, $userGroupsIds)) {
+                    array_push($resourceCapabilities, $resource['capabilities']);
```

Comment:
I think we might get duplicated capabilities in case the same capability is assigned to two different resources. Is that correct case?